### ISSUE TITLE
Provide OVF support

### ISSUE THREAD

**Issue by bigfoot1942** [choderalab/coronavirus#36]
If you were to offer an OVF containing a virtual appliance that can use nVIDIA Tesla T4 videocards, using vGPU, that would greatly reduce effort to implement this in an enterprise environment.

**Comment by jchodera** [comment]
Great idea!

@jaimergp : Any chance you can look into this?

**Comment by jaimergp** [comment]
I have only tangential experience with OVF/OVA files, but I am happy to help. I have some questions though:

* Wouldn't we have license issues if we repackage things such as the Nvidia drivers or CUDA binaries?
* What's the expected OS for the OVA? I guess a Linux flavor?
* Wouldn't Docker images serve the same purpose here? Is there any chance you can use those to help you create the OVF/OVA? There's also Azure templates.

**Comment by bigfoot1942** [comment]
If you were to use docker - I guess VMware would advise you to use PhotonOS
to run it in.
But then I'm not sure on how to configure vGPU inside that...
This would basically be the on-prem alternative to using Azure (where the
PAYG model might not help the cause)

If licensing were to be an issue I guess it might be wise to contact nVidia
as they might be willing to hand out a waiver this specific project.

**Comment by jaimergp** [comment]
Ok, I have tried my best to export amalic's Docker image to an OVF/OVA appliance. You can get it here: https://we.tl/t-fn2IoAt8LZ

I followed these steps:

1. Cloned https://github.com/iximiuz/docker-to-linux and edited [the Ubuntu recipe](https://github.com/iximiuz/docker-to-linux/blob/master/ubuntu/Dockerfile) so it uses [amalic's docker image](https://hub.docker.com/r/amalic/nvdocker-folding-home) instead.
2. Added `vi`, `nano` and `network-manager` to the apt install instructions.
3. `make ubuntu`
4. `qemu-img convert -i raw -O vmdk linux.img fah.vmdk`
5. Created a new VirtualBox VM using that VMDK file. At this point you can login with `root:root`.
6. Patched `/etc/fstab` to avoid read-only bugs.
7. Patched `/etc/init.d/FAHClient` so it uses the `root`  user (otherwise it fails to start).
8. Exported the whole thing as an OVA appliance.

Basically, FAHClient will start in daemon mode even if you don't login, so that's all set up. You might want to edit `/etc/fahclient/config.xml` to specify username/team and/or enable more slots.

BUT:

* I don't have access to VMWare to test it.
* I haven't tried to expose the GPU to the VM.
* I don't have vGPU-capable hardware so I can't test it either.

Hopefully this is enough to get you started?

**Comment by bigfoot1942** [comment]
Is this some parallel work?

https://twitter.com/lamw/status/1240656809796362241?s=19

**Comment by jaimergp** [comment]
I am not aware of any more efforts being done, but given the popularity of F@H at this moment, I wouldn't be surprised! @seanpmassey, can you offer some more details here? Thanks!

**Comment by bigfoot1942** [comment]
It looks like VMware is building this in an internal project and making it
ready to deploy and they will make the final project publicly accessible.
  https://twitter.com/lamw check his last posts.

**Comment by jaimergp** [comment]
Awesome, thanks for the report!

**Comment by jaimergp** [comment]
For those wondering, OVA support is already here:

https://flings.vmware.com/vmware-appliance-for-folding-home
